src/templates/plugin.main.js

```diff
@@ -185,6 +186,10 @@ export default async (context) => {
       return false
     }
 
+    if (route.path.match(getLocalesRegex(localeCodes)) && !alwaysRedirect) {
+      return false
+    }
```

Comment:
There is a `getLocaleFromRoute` function available in this file. It would be better to use that than introducing duplicate code for parsing locale. `getLocaleFromRoute` gets locale from route's **name** and not **path** but I think that would be OK and more consistent.

Reply:
It would probably be even more correct in case of using the default `prefix_except_default` strategy.

In that case, the default locale is unprefixed and your code, when visiting `/foo` for example, would trigger redirect while using `getLocaleFromRoute` shouldn't.

So it depends on what are your expectations but in any case, we have to think about how it's gonna affect every strategy.

Reply:
@rchl Isn't this exactly what I did originally in my first commit?
https://github.com/nuxt-community/i18n-module/pull/896/commits/1110aafd4f0f818f724226b34065c05ea37a8ee4

Reply:
Yes, sorry. Hard to keep track when reviewing over a couple of months. :)